lib/qs/client.rb

```diff
@@ -49,8 +49,8 @@ def initialize(*args)
     private
 
     def enqueue!(queue, job)
-      encoded_payload = Qs::Payload.encode(job.to_payload)
-      self.redis.with{ |c| c.lpush(queue.redis_key, encoded_payload) }
+      serialized_payload = self.serializer.call(job.to_payload)
```

Comment:
@jcredding I'll comment further in a bit, but I'd prefer to keep this `Payload` module and encode function and not have to pass in and store a serializer on the clients.